File: 2023/01 - January/26/03/Artist.java
// Artist.java

// Represents An Artist
public class Artist {
	// Declaring Required Variables
	private String artistName;
	private int birthYear;
	private int deathYear;

	// Default Constructor
	public Artist() {
		artistName = "unknown";
		birthYear = -1;
		deathYear = -1;
	}


	// Artist Constructor
	public Artist(String artistName, int birthYear, int deathYear) {
		this.artistName = artistName;
		this.birthYear = birthYear;
		this.deathYear = deathYear;
	}

	// Getters For Name
	public String getName() {
		return artistName;
	}

	// Getters For BirthYear
	public int getBirthYear() {
		return birthYear;
	}

	// Getters For DeathYear
	public int getDeathYear() {
		return deathYear;
	}

	// Printing Information
	public void printInfo() {
		// Death Year And Birth Year, Both Are Known
		if (deathYear != -1 && birthYear != -1) {
			System.out.printf("Artist: %s (%d to %d)\n",
				artistName, birthYear, deathYear);
		// Only Birth Year Is Known (Artist Is Alive)
		} else if (birthYear != -1) {
			System.out.printf("Artist: %s (%d to present)\n",
				artistName, birthYear);
		// Neither Of BirthYear And DeathYear Are Known
		} else {
			System.out.printf("Artist: %s (unknown)\n",
				artistName);
		}
	}
}

File: 2023/02 - February/06/04/Artist.cpp
#include "Artist.h"

// Default Constructor
// Initializes Name = Unkown, Birth Year And Death Year To -1
Artist::Artist() : Artist("unknown", -1, -1) {}

// Parametrized Constructor
Artist::Artist(std::string artistName, int birthYear, int deathYear) {
	this->artistName = artistName;
	this->birthYear = birthYear;
	this->deathYear = deathYear;
}

// Getter For Name
std::string Artist::GetName() const {
	return artistName;
}


// Getter For Death Year
int Artist::GetDeathYear() const {
	return deathYear;
}


// Getter For Birth Year
int Artist::GetBirthYear() const {
	return birthYear;
}


// Prints Artist Info
void Artist::PrintInfo() const {
	std::cout << "Artist: " << artistName << " (";

	// If Birth Year And Death Year Both Are Known
	if (birthYear != -1 && deathYear != -1) {
		std::cout << birthYear << " to " << deathYear;
	// If Death Year Is Not Known (Artist Is Still Alive)
	} else if (birthYear != -1) {
		std::cout << birthYear << " to present";
	// If Neither Of Birth Year And Death Year Are Known
	} else {
		std::cout << "unkown";
	}
	std::cout << ")" << std::endl;
}
